liquidctl/driver/coolit.py

```diff
@@ -0,0 +1,313 @@
+            ('Fan 1 speed', u16le_from(res, offset=8), 'rpm'),
+            ('Fan 2 speed', u16le_from(res, offset=14), 'rpm'),
+            ('Pump speed', u16le_from(res, offset=20), 'rpm'),
+        ]
+
+    def set_fixed_speed(self, channel, duty, **kwargs):
+        """Set fan or fans to a fixed speed duty.
+
+        Valid channel values are 'fanN', where N >= 1 is the fan number, and
+        'fan', to simultaneously configure all fans.  Unconfigured fan channels
+        may default to 100% duty.
+        """
+        for hw_channel in self._get_hw_fan_channels(channel):
+            self._data.store(f'{hw_channel}_mode', _FanMode.FIXED_DUTY.value)
+            self._data.store(f'{hw_channel}_duty', duty)
+        self._send_set_cooling()
+
+    def set_speed_profile(self, channel, profile, **kwargs):
+        """Set fan or fans to follow a speed duty profile.
+
+        Valid channel values are 'fanN', where N >= 1 is the fan number, and
+        'fan', to simultaneously configure all fans.  Unconfigured fan channels
+        may default to 100% duty.
+
+        Up to seven (temperature, duty) pairs can be supplied in `profile`,
+        with temperatures in Celsius and duty values in percentage.  The last
+        point should set the fan to 100% duty cycle, or be omitted; in the
+        latter case the fan will be set to max out at 60°C.
+        """
+        profile = list(profile)
+        for hw_channel in self._get_hw_fan_channels(channel):
+            self._data.store(f'{hw_channel}_mode', _FanMode.CUSTOM_PROFILE.value)
+            self._data.store(f'{hw_channel}_profile', profile)
+        self._send_set_cooling()
+
+    def _get_hw_fan_channels(self, channel):
+        channel = channel.lower()
+        if channel == 'fan':
+            return self._fan_names
+        if channel in self._fan_names:
+            return [channel]
+        raise ValueError(f'Unknown channel, should be one of: {_quoted("fan", *self._fan_names)}')
+
+    def _build_data_package(self, command, opCode, params=None):
+        if params:
+            buf = bytearray(3 + len(params))
+            buf[3 : 3 + len(params)] = params
+        else:
+            buf = bytearray(3)
+
+        buf[0] = next(self._sequence)
+        buf[1] = opCode
+        buf[2] = command
+
+        return buf
+
+    def _send_command(self, dataPackages):
+        buf = bytearray(_REPORT_LENGTH)
+
+        startIndex = 1
+        for dataPackage in dataPackages:
+            buf[startIndex : startIndex + len(dataPackage)] = dataPackage
+            startIndex += len(dataPackage)
+
+        buf[0] = startIndex - 1
+        
+        LOGGER.debug('write %s', buf.hex())
+        self.device.clear_enqueued_reports()
+        self.device.write(buf)
+        buf = bytes(self.device.read(_REPORT_LENGTH))
+        self.device.release()
+        LOGGER.debug('received %s', buf.hex())
+        return buf
+
+    def _send_set_cooling(self):
```

Comment:
This can be simplified, since the protocol doesn't require all cooling settings to be sent together.

In this case sending the commands immediately, instead of saving them to temporary storage (`self._data`) and sending them in groups (what this function does), is cheaper and bit simpler.